src/pages/SingleCourse.tsx

```diff
@@ -866,9 +872,28 @@ const SingleCourse: React.FC = () => {
         { variables: { subcourseId } }
     );
 
+    const [promote, _promote] = useMutation(
```

Comment:
You can just use destructuring here
```suggestion
    const [promote, { data, loading, error }] = useMutation(
```